wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1696,6 +1696,15 @@ impl super::Instance {
             );
         };
 
+        if info.driver_info.contains("Mesa ") {
```

Comment:
From meeting: decide if we should narrow this condition.

Reply:
I narrowed it to look for `driver == "llvmpipe"`.

I also noticed that the gles and vulkan backends aren't written to set newly-defined downlevel flags by default, maybe they should be? And regardless of that, should the gles backend enable F16_IN_F32?